src/main/webapp/app/exam/manage/exams/exam-detail.component.ts

```diff
@@ -16,15 +20,38 @@ export class ExamDetailComponent implements OnInit {
     formattedEndText?: SafeHtml;
     formattedConfirmationEndText?: SafeHtml;
     isAtLeastInstructor = false;
+    isLoading = false;
+    pointsExercisesEqual = false;
+    allExamsGenerated = false;
+    allGroupsContainExercise = false;
+    totalPointsMandatory = false;
+    totalPointsMandatoryOptional = false;
 
-    constructor(private route: ActivatedRoute, private artemisMarkdown: ArtemisMarkdownService, private accountService: AccountService) {}
+    constructor(
+        private route: ActivatedRoute,
+        private artemisMarkdown: ArtemisMarkdownService,
+        private accountService: AccountService,
+        private examService: ExamManagementService,
+        private exerciseGroupService: ExerciseGroupService,
+    ) {}
 
     /**
      * Initialize the exam
      */
     ngOnInit(): void {
+        this.isLoading = true;
         this.route.data.subscribe(({ exam }) => {
             this.exam = exam;
+            this.exerciseGroupService
+                .findAllForExam(this.exam!.course!.id!, this.exam.id!)
+                .map((exerciseGroupArray: HttpResponse<ExerciseGroup[]>) => exerciseGroupArray.body!)
+                .subscribe((x) => {
+                    this.exam.exerciseGroups = x;
```

Comment:
a better variable name would be nice ;) 

```suggestion
                .subscribe((exGroups) => {
                    this.exam.exerciseGroups = exGroups;
```